ersilia/utils/identifiers/compound.py

```diff
@@ -122,25 +124,46 @@ def chemical_identifier_resolver(identifier):
 
     def encode(self, smiles):
         """Get InChIKey of compound based on SMILES string"""
-        if self.Chem is None:
-            inchikey = self._pubchem_smiles_to_inchikey(smiles)
-            if inchikey is None:
-                inchikey = self._nci_smiles_to_inchikey(smiles)
-        else:
+        if smiles is None or not smiles.strip():
+            return self.UNPROCESSABLE_INPUT
+        
+        inchikey = None
+
+        if self.Chem is not None:
             try:
                 mol = self.Chem.MolFromSmiles(smiles)
-                if mol is None:
-                    raise Exception(
-                        "The SMILES string: %s is not valid or could not be converted to an InChIKey"
-                        % smiles
-                    )
-                inchi = self.Chem.rdinchi.MolToInchi(mol)[0]
-                if inchi is None:
-                    raise Exception("Could not obtain InChI")
-                inchikey = self.Chem.rdinchi.InchiToInchiKey(inchi)
+                if mol is not None:
+                    inchi = self.Chem.rdinchi.MolToInchi(mol)[0]
+                    inchikey = self.Chem.rdinchi.InchiToInchiKey(inchi)
             except:
-                inchikey = None
-        return inchikey
+                pass
 
+        if inchikey is None:
+            inchikey = self._pubchem_smiles_to_inchikey(smiles)
+
+        if inchikey is None:
+            inchikey = self._nci_smiles_to_inchikey(smiles)
+
+        return inchikey or self.UNPROCESSABLE_INPUT
+
+    
+    def process_input(self, smiles_list):
```

Comment:
Where is the caller for this method? I don't think this is being called anywhere in the code.

Reply:
the class is instatiated here
https://github.com/ersilia-os/ersilia/blob/43bbe1d298217126987268495f0d831ddfa16d08/ersilia/io/types/compound.py#L26

where the method i guess should handle the process input .
while looking at the instatiation, it seems the  self.identifier reference object of the  CompoundIdentifier  class has no methods
The importlib seems not to import the class